code/modules/mob/living/simple_animal/abnormality/teth/fragment.dm

```diff
@@ -81,16 +91,42 @@
 	if(song_cooldown > world.time)
 		return
 	can_act = FALSE
+	flick("fragment_song_start" , src)
+	SLEEP_CHECK_DEATH(5)
+	legs = new(get_turf(src))
+	icon_state = "fragment_song_head"
+	pixel_y = 5
+	particle_notes = new(get_turf(src))
+	particle_notes.pixel_y = 26
+	particle_song = new(get_turf(src))
+	particle_song.pixel_y = 26
 	playsound(get_turf(src), 'sound/abnormalities/fragment/sing.ogg', 50, 0, 4)
 	for(var/i = 1 to 8)
-		new /obj/effect/temp_visual/fragment_song(get_turf(src))
+
+		//Animation for bobbing the head left to right
+		switch(i)
+			if(1)
+				animate(src, transform = turn(matrix(), -90), time = 6, flags = SINE_EASING | EASE_OUT )
+			if(3)
+				animate(src, transform = turn(matrix(), 0), time = 6, flags = SINE_EASING | EASE_IN | EASE_OUT )
+			if(5)
+				animate(src, transform = turn(matrix(), 90), time = 6, flags = SINE_EASING | EASE_IN | EASE_OUT )
+			if(7)
+				animate(src, transform = turn(matrix(), 0), time = 6, flags = SINE_EASING | EASE_IN )
+
 		for(var/mob/living/L in view(8, src))
 			if(faction_check_mob(L, FALSE))
 				continue
 			if(L.stat == DEAD)
 				continue
 			L.apply_damage(song_damage, WHITE_DAMAGE, null, L.run_armor_check(null, WHITE_DAMAGE), spread_damage = TRUE)
 		SLEEP_CHECK_DEATH(3)
+	animate(src, pixel_y = 0, time = 0)
+	icon_state = "fragment_breach"
+	pixel_y = 0
+	QDEL_NULL(legs)
+	QDEL_NULL(particle_notes)
+	QDEL_NULL(particle_song)
```

Comment:
Don't qdel the particles, let them naturally fade. It should look better that way.

Reply:
Done, added fadeout